src/routes/privateSnotra.ts

```diff
@@ -0,0 +1,60 @@
+import { FastifyInstance } from 'fastify';
+import createOrGetConnection from '../db';
+import { Keyword, Source, Post } from '../entity';
+
+export default async function (fastify: FastifyInstance): Promise<void> {
+  fastify.get('/all_tags', async (req, res) => {
+    if (!req.service) {
+      return res.status(400).send();
+    }
+
+    const con = await createOrGetConnection();
+    const result = await con
+      .createQueryBuilder()
+      .select('value', 'name')
+      .from(Keyword, 'keyword')
+      .where('status = :status', { status: 'allow' })
+      .orderBy('name', 'ASC')
+      .getRawMany<{ name: string }>();
+
+    return res.status(200).send(result);
+  });
+
+  fastify.get('/all_sources', async (req, res) => {
+    if (!req.service) {
+      return res.status(400).send();
+    }
+
+    const con = await createOrGetConnection();
+    const result = await con
+      .createQueryBuilder()
+      .select('id', 'name')
+      .from(Source, 'source')
+      .where('type = :type', { type: 'machine' })
+      .andWhere('private = :private', { private: false })
+      .andWhere('active = :active', { active: true })
+      .orderBy('name', 'ASC')
+      .getRawMany<{ name: string }>();
+
+    return res.status(200).send(result);
+  });
+
+  fastify.get('/all_content_curations', async (req, res) => {
+    if (!req.service) {
+      return res.status(400).send();
+    }
+
+    const con = await createOrGetConnection();
+    // this query might be hard on a database, we can consider extracting content curations
+    // to a separate table
+    const result = await con
```

Comment:
@idoshamun You happen to know any optimisations here?
Not sure how often this endpoint will be called/implications.,

Reply:
@mykytanikitenko not ideal, as you rightfully pointed, let's add a filter to all posts of the last 30 days or so. Should give us enough. Another option is to use static options for now. Anyway our NLP method returns fixed options. It's very structured

Reply:
I added the last 30 days. Honestly, I would prefer query to static options. Once a day during the night, it shouldn't hit the db very hard :)